// @streampay/checkout-sdk/dist/index.js
var A = Object.create;
var n = Object.defineProperty,
    v = Object.defineProperties,
    W = Object.getOwnPropertyDescriptor,
    q = Object.getOwnPropertyDescriptors,
    T = Object.getOwnPropertyNames,
    f = Object.getOwnPropertySymbols,
    L = Object.getPrototypeOf,
    l = Object.prototype.hasOwnProperty,
    U = Object.prototype.propertyIsEnumerable;
var u = (t, e, o) => e in t ? n(t, e, { enumerable: !0, configurable: !0, writable: !0, value: o }) : t[e] = o,
    k = (t, e) => {
        for (var o in e || (e = {})) l.call(e, o) && u(t, o, e[o]);
        if (f)
            for (var o of f(e)) U.call(e, o) && u(t, o, e[o]);
        return t
    },
    w = (t, e) => v(t, q(e)),
    r = (t, e) => n(t, "name", { value: e, configurable: !0 });
var _ = (t, e) => { for (var o in e) n(t, o, { get: e[o], enumerable: !0 }) },
    g = (t, e, o, s) => {
        if (e && typeof e == "object" || typeof e == "function")
            for (let i of T(e)) !l.call(t, i) && i !== o && n(t, i, { get: () => e[i], enumerable: !(s = W(e, i)) || s.enumerable });
        return t
    };
var P = (t, e, o) => (o = t != null ? A(L(t)) : {}, g(e || !t || !t.__esModule ? n(o, "default", { value: t, enumerable: !0 }) : o, t)),
    M = t => g(n({}, "__esModule", { value: !0 }), t);
var D = {};
_(D, { StreamPay: () => h, getSupportedTokenMintAddress: () => O, preciseRound: () => m });
module.exports = M(D);
var x = "https://checkout-production.streampayment.app/api/v1";
var b = new Map([
    ["sol", "So11111111111111111111111111111111111111112"],
    ["usdc", "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v"],
    ["str", "5P3giWpPBrVKL8QP8roKM7NsLdi3ie1Nc2b5r9mGtvwb"],
    ["samo", "7xKXtg2CW87d97TXJSDpbD5jBkheTqA83TZRuJosgAsU"],
    ["shdw", "SHDWyBxihqiCj6YekG2GUr7wqKLeLAMK1gHZck9pL6y"]
]);
var O = r(t => b.get(t), "getSupportedTokenMintAddress"),
    m = r((t, e) => { let o = e ? Math.pow(10, e) : 1; return Math.ceil(t * o) / o }, "preciseRound");
var p = class {
    constructor(e, o, s) { this.client = e, this.network = o, this.config = s }
    async create(e) {
        try {
            let o = m(e.items.reduce((y, { price: R, quantity: C }) => y + R * C, 0), 3);
            e.shipping_fees && (o = m(o + e.shipping_fees, 3));
            let s = w(k({}, e), { network: this.network, amount: o, config: this.config });
            return (await this.client.post("/session", s)).data
        } catch (o) { throw new Error(o) }
    }
};
r(p, "Session");
var d = require("svix");
var c = class {
    constructor(e) { this.client = e }
    async register(e) { try { return (await this.client.post("/webhook", e)).data } catch (o) { throw new Error(o) } }
    async verify(e) { let o = new d.Webhook(e.webhook_secret); try { await o.verify(e.payload, e.headers) } catch (s) { throw new Error("Invalid webhook signature") } }
};
r(c, "Webhook");
var S = P(require("axios"));
var a = class {
    constructor(e) { this.request = S.default.create({ baseURL: x, headers: { Authorization: `Bearer ${e.api_key}` } }) }
    async get(e, o) { let s = "?" + new URLSearchParams(o).toString(); return this.request.get(`${e}?${s}`) }
    async post(e, o) { return this.request.post(`${e}`, o) }
};
r(a, "Client");
var h = class { constructor(e) { this.client = new a(e), this.webhook = new c(this.client), this.session = new p(this.client, e.network, e.config) } };
r(h, "SteamPay");
0 && (module.exports = { StreamPay, getSupportedTokenMintAddress, preciseRound });
