// svix/dist/openapi/types/PromiseAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromiseStatisticsApi = exports.PromiseMessageAttemptApi = exports.PromiseMessageApi = exports.PromiseIntegrationApi = exports.PromiseHealthApi = exports.PromiseEventTypeApi = exports.PromiseEnvironmentSettingsApi = exports.PromiseEnvironmentApi = exports.PromiseEndpointApi = exports.PromiseAuthenticationApi = exports.PromiseApplicationApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class PromiseApplicationApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableApplicationApi(configuration, requestFactory, responseProcessor);
    }
    createApplicationApiV1AppPost(applicationIn, getIfExists, idempotencyKey, _options) {
        const result = this.api.createApplicationApiV1AppPost(applicationIn, getIfExists, idempotencyKey, _options);
        return result.toPromise();
    }
    deleteApplicationApiV1AppAppIdDelete(appId, idempotencyKey, _options) {
        const result = this.api.deleteApplicationApiV1AppAppIdDelete(appId, idempotencyKey, _options);
        return result.toPromise();
    }
    getAppStatsApiV1AppAppIdStatsGet(appId, since, until, idempotencyKey, _options) {
        const result = this.api.getAppStatsApiV1AppAppIdStatsGet(appId, since, until, idempotencyKey, _options);
        return result.toPromise();
    }
    getAppUsageStatsApiV1AppStatsUsageGet(since, until, limit, iterator, idempotencyKey, _options) {
        const result = this.api.getAppUsageStatsApiV1AppStatsUsageGet(since, until, limit, iterator, idempotencyKey, _options);
        return result.toPromise();
    }
    getApplicationApiV1AppAppIdGet(appId, idempotencyKey, _options) {
        const result = this.api.getApplicationApiV1AppAppIdGet(appId, idempotencyKey, _options);
        return result.toPromise();
    }
    listApplicationsApiV1AppGet(iterator, limit, idempotencyKey, _options) {
        const result = this.api.listApplicationsApiV1AppGet(iterator, limit, idempotencyKey, _options);
        return result.toPromise();
    }
    updateApplicationApiV1AppAppIdPut(appId, applicationIn, idempotencyKey, _options) {
        const result = this.api.updateApplicationApiV1AppAppIdPut(appId, applicationIn, idempotencyKey, _options);
        return result.toPromise();
    }
}
exports.PromiseApplicationApi = PromiseApplicationApi;
const ObservableAPI_2 = require("./ObservableAPI");
class PromiseAuthenticationApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableAuthenticationApi(configuration, requestFactory, responseProcessor);
    }
    getDashboardAccessApiV1AuthDashboardAccessAppIdPost(appId, idempotencyKey, _options) {
        const result = this.api.getDashboardAccessApiV1AuthDashboardAccessAppIdPost(appId, idempotencyKey, _options);
        return result.toPromise();
    }
    logoutApiV1AuthLogoutPost(idempotencyKey, _options) {
        const result = this.api.logoutApiV1AuthLogoutPost(idempotencyKey, _options);
        return result.toPromise();
    }
}
exports.PromiseAuthenticationApi = PromiseAuthenticationApi;
const ObservableAPI_3 = require("./ObservableAPI");
class PromiseEndpointApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_3.ObservableEndpointApi(configuration, requestFactory, responseProcessor);
    }
    createEndpointApiV1AppAppIdEndpointPost(appId, endpointIn, idempotencyKey, _options) {
        const result = this.api.createEndpointApiV1AppAppIdEndpointPost(appId, endpointIn, idempotencyKey, _options);
        return result.toPromise();
    }
    deleteEndpointApiV1AppAppIdEndpointEndpointIdDelete(endpointId, appId, idempotencyKey, _options) {
        const result = this.api.deleteEndpointApiV1AppAppIdEndpointEndpointIdDelete(endpointId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    getEndpointApiV1AppAppIdEndpointEndpointIdGet(endpointId, appId, idempotencyKey, _options) {
        const result = this.api.getEndpointApiV1AppAppIdEndpointEndpointIdGet(endpointId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    getEndpointHeadersApiV1AppAppIdEndpointEndpointIdHeadersGet(endpointId, appId, idempotencyKey, _options) {
        const result = this.api.getEndpointHeadersApiV1AppAppIdEndpointEndpointIdHeadersGet(endpointId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    getEndpointSecretApiV1AppAppIdEndpointEndpointIdSecretGet(endpointId, appId, idempotencyKey, _options) {
        const result = this.api.getEndpointSecretApiV1AppAppIdEndpointEndpointIdSecretGet(endpointId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    getEndpointStatsApiV1AppAppIdEndpointEndpointIdStatsGet(endpointId, appId, idempotencyKey, _options) {
        const result = this.api.getEndpointStatsApiV1AppAppIdEndpointEndpointIdStatsGet(endpointId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    getEndpointTransformationApiV1AppAppIdEndpointEndpointIdTransformationGet(endpointId, appId, idempotencyKey, _options) {
        const result = this.api.getEndpointTransformationApiV1AppAppIdEndpointEndpointIdTransformationGet(endpointId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    listEndpointsApiV1AppAppIdEndpointGet(appId, iterator, limit, idempotencyKey, _options) {
        const result = this.api.listEndpointsApiV1AppAppIdEndpointGet(appId, iterator, limit, idempotencyKey, _options);
        return result.toPromise();
    }
    patchEndpointHeadersApiV1AppAppIdEndpointEndpointIdHeadersPatch(appId, endpointId, endpointHeadersPatchIn, idempotencyKey, _options) {
        const result = this.api.patchEndpointHeadersApiV1AppAppIdEndpointEndpointIdHeadersPatch(appId, endpointId, endpointHeadersPatchIn, idempotencyKey, _options);
        return result.toPromise();
    }
    recoverFailedWebhooksApiV1AppAppIdEndpointEndpointIdRecoverPost(appId, endpointId, recoverIn, idempotencyKey, _options) {
        const result = this.api.recoverFailedWebhooksApiV1AppAppIdEndpointEndpointIdRecoverPost(appId, endpointId, recoverIn, idempotencyKey, _options);
        return result.toPromise();
    }
    replayMissingWebhooksApiV1AppAppIdEndpointEndpointIdReplayMissingPost(appId, endpointId, replayIn, idempotencyKey, _options) {
        const result = this.api.replayMissingWebhooksApiV1AppAppIdEndpointEndpointIdReplayMissingPost(appId, endpointId, replayIn, idempotencyKey, _options);
        return result.toPromise();
    }
    rotateEndpointSecretApiV1AppAppIdEndpointEndpointIdSecretRotatePost(endpointId, appId, endpointSecretRotateIn, idempotencyKey, _options) {
        const result = this.api.rotateEndpointSecretApiV1AppAppIdEndpointEndpointIdSecretRotatePost(endpointId, appId, endpointSecretRotateIn, idempotencyKey, _options);
        return result.toPromise();
    }
    sendEventTypeExampleMessageApiV1AppAppIdEndpointEndpointIdSendExamplePost(appId, endpointId, eventExampleIn, idempotencyKey, _options) {
        const result = this.api.sendEventTypeExampleMessageApiV1AppAppIdEndpointEndpointIdSendExamplePost(appId, endpointId, eventExampleIn, idempotencyKey, _options);
        return result.toPromise();
    }
    setEndpointTransformationApiV1AppAppIdEndpointEndpointIdTransformationPatch(appId, endpointId, endpointTransformationIn, idempotencyKey, _options) {
        const result = this.api.setEndpointTransformationApiV1AppAppIdEndpointEndpointIdTransformationPatch(appId, endpointId, endpointTransformationIn, idempotencyKey, _options);
        return result.toPromise();
    }
    simulateEndpointTransformationApiV1AppAppIdEndpointEndpointIdTransformationSimulatePost(endpointId, appId, endpointTransformationSimulateIn, idempotencyKey, _options) {
        const result = this.api.simulateEndpointTransformationApiV1AppAppIdEndpointEndpointIdTransformationSimulatePost(endpointId, appId, endpointTransformationSimulateIn, idempotencyKey, _options);
        return result.toPromise();
    }
    updateEndpointApiV1AppAppIdEndpointEndpointIdPut(endpointId, appId, endpointUpdate, idempotencyKey, _options) {
        const result = this.api.updateEndpointApiV1AppAppIdEndpointEndpointIdPut(endpointId, appId, endpointUpdate, idempotencyKey, _options);
        return result.toPromise();
    }
    updateEndpointHeadersApiV1AppAppIdEndpointEndpointIdHeadersPut(appId, endpointId, endpointHeadersIn, idempotencyKey, _options) {
        const result = this.api.updateEndpointHeadersApiV1AppAppIdEndpointEndpointIdHeadersPut(appId, endpointId, endpointHeadersIn, idempotencyKey, _options);
        return result.toPromise();
    }
}
exports.PromiseEndpointApi = PromiseEndpointApi;
const ObservableAPI_4 = require("./ObservableAPI");
class PromiseEnvironmentApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_4.ObservableEnvironmentApi(configuration, requestFactory, responseProcessor);
    }
    exportEnvironmentConfigurationApiV1EnvironmentExportPost(body, idempotencyKey, _options) {
        const result = this.api.exportEnvironmentConfigurationApiV1EnvironmentExportPost(body, idempotencyKey, _options);
        return result.toPromise();
    }
    importEnvironmentConfigurationApiV1EnvironmentImportPost(environmentIn, idempotencyKey, _options) {
        const result = this.api.importEnvironmentConfigurationApiV1EnvironmentImportPost(environmentIn, idempotencyKey, _options);
        return result.toPromise();
    }
}
exports.PromiseEnvironmentApi = PromiseEnvironmentApi;
const ObservableAPI_5 = require("./ObservableAPI");
class PromiseEnvironmentSettingsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_5.ObservableEnvironmentSettingsApi(configuration, requestFactory, responseProcessor);
    }
    getOrgSettingsApiV1EnvironmentSettingsGet(idempotencyKey, _options) {
        const result = this.api.getOrgSettingsApiV1EnvironmentSettingsGet(idempotencyKey, _options);
        return result.toPromise();
    }
}
exports.PromiseEnvironmentSettingsApi = PromiseEnvironmentSettingsApi;
const ObservableAPI_6 = require("./ObservableAPI");
class PromiseEventTypeApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_6.ObservableEventTypeApi(configuration, requestFactory, responseProcessor);
    }
    createEventTypeApiV1EventTypePost(eventTypeIn, idempotencyKey, _options) {
        const result = this.api.createEventTypeApiV1EventTypePost(eventTypeIn, idempotencyKey, _options);
        return result.toPromise();
    }
    deleteEventTypeApiV1EventTypeEventTypeNameDelete(eventTypeName, idempotencyKey, _options) {
        const result = this.api.deleteEventTypeApiV1EventTypeEventTypeNameDelete(eventTypeName, idempotencyKey, _options);
        return result.toPromise();
    }
    generateSchemaExampleApiV1EventTypeSchemaGenerateExamplePost(eventTypeSchemaIn, idempotencyKey, _options) {
        const result = this.api.generateSchemaExampleApiV1EventTypeSchemaGenerateExamplePost(eventTypeSchemaIn, idempotencyKey, _options);
        return result.toPromise();
    }
    getEventTypeApiV1EventTypeEventTypeNameGet(eventTypeName, idempotencyKey, _options) {
        const result = this.api.getEventTypeApiV1EventTypeEventTypeNameGet(eventTypeName, idempotencyKey, _options);
        return result.toPromise();
    }
    listEventTypesApiV1EventTypeGet(iterator, limit, withContent, includeArchived, idempotencyKey, _options) {
        const result = this.api.listEventTypesApiV1EventTypeGet(iterator, limit, withContent, includeArchived, idempotencyKey, _options);
        return result.toPromise();
    }
    updateEventTypeApiV1EventTypeEventTypeNamePut(eventTypeName, eventTypeUpdate, idempotencyKey, _options) {
        const result = this.api.updateEventTypeApiV1EventTypeEventTypeNamePut(eventTypeName, eventTypeUpdate, idempotencyKey, _options);
        return result.toPromise();
    }
}
exports.PromiseEventTypeApi = PromiseEventTypeApi;
const ObservableAPI_7 = require("./ObservableAPI");
class PromiseHealthApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_7.ObservableHealthApi(configuration, requestFactory, responseProcessor);
    }
    healthApiV1HealthGet(idempotencyKey, _options) {
        const result = this.api.healthApiV1HealthGet(idempotencyKey, _options);
        return result.toPromise();
    }
}
exports.PromiseHealthApi = PromiseHealthApi;
const ObservableAPI_8 = require("./ObservableAPI");
class PromiseIntegrationApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_8.ObservableIntegrationApi(configuration, requestFactory, responseProcessor);
    }
    createIntegrationApiV1AppAppIdIntegrationPost(appId, integrationIn, idempotencyKey, _options) {
        const result = this.api.createIntegrationApiV1AppAppIdIntegrationPost(appId, integrationIn, idempotencyKey, _options);
        return result.toPromise();
    }
    deleteIntegrationApiV1AppAppIdIntegrationIntegIdDelete(integId, appId, idempotencyKey, _options) {
        const result = this.api.deleteIntegrationApiV1AppAppIdIntegrationIntegIdDelete(integId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    getIntegrationApiV1AppAppIdIntegrationIntegIdGet(integId, appId, idempotencyKey, _options) {
        const result = this.api.getIntegrationApiV1AppAppIdIntegrationIntegIdGet(integId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    getIntegrationKeyApiV1AppAppIdIntegrationIntegIdKeyGet(integId, appId, idempotencyKey, _options) {
        const result = this.api.getIntegrationKeyApiV1AppAppIdIntegrationIntegIdKeyGet(integId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    listIntegrationsApiV1AppAppIdIntegrationGet(appId, iterator, limit, idempotencyKey, _options) {
        const result = this.api.listIntegrationsApiV1AppAppIdIntegrationGet(appId, iterator, limit, idempotencyKey, _options);
        return result.toPromise();
    }
    rotateIntegrationKeyApiV1AppAppIdIntegrationIntegIdKeyRotatePost(integId, appId, idempotencyKey, _options) {
        const result = this.api.rotateIntegrationKeyApiV1AppAppIdIntegrationIntegIdKeyRotatePost(integId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    updateIntegrationApiV1AppAppIdIntegrationIntegIdPut(integId, appId, integrationUpdate, idempotencyKey, _options) {
        const result = this.api.updateIntegrationApiV1AppAppIdIntegrationIntegIdPut(integId, appId, integrationUpdate, idempotencyKey, _options);
        return result.toPromise();
    }
}
exports.PromiseIntegrationApi = PromiseIntegrationApi;
const ObservableAPI_9 = require("./ObservableAPI");
class PromiseMessageApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_9.ObservableMessageApi(configuration, requestFactory, responseProcessor);
    }
    createMessageApiV1AppAppIdMsgPost(appId, messageIn, withContent, idempotencyKey, _options) {
        const result = this.api.createMessageApiV1AppAppIdMsgPost(appId, messageIn, withContent, idempotencyKey, _options);
        return result.toPromise();
    }
    getMessageApiV1AppAppIdMsgMsgIdGet(msgId, appId, idempotencyKey, _options) {
        const result = this.api.getMessageApiV1AppAppIdMsgMsgIdGet(msgId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    listMessagesApiV1AppAppIdMsgGet(appId, iterator, limit, eventTypes, channel, before, after, idempotencyKey, _options) {
        const result = this.api.listMessagesApiV1AppAppIdMsgGet(appId, iterator, limit, eventTypes, channel, before, after, idempotencyKey, _options);
        return result.toPromise();
    }
}
exports.PromiseMessageApi = PromiseMessageApi;
const ObservableAPI_10 = require("./ObservableAPI");
class PromiseMessageAttemptApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_10.ObservableMessageAttemptApi(configuration, requestFactory, responseProcessor);
    }
    getAttemptApiV1AppAppIdMsgMsgIdAttemptAttemptIdGet(attemptId, msgId, appId, idempotencyKey, _options) {
        const result = this.api.getAttemptApiV1AppAppIdMsgMsgIdAttemptAttemptIdGet(attemptId, msgId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
    getAttemptHeadersApiV1AppAppIdMsgMsgIdAttemptAttemptIdHeadersGet(appId, msgId, attemptId, idempotencyKey, _options) {
        const result = this.api.getAttemptHeadersApiV1AppAppIdMsgMsgIdAttemptAttemptIdHeadersGet(appId, msgId, attemptId, idempotencyKey, _options);
        return result.toPromise();
    }
    listAttemptedDestinationsApiV1AppAppIdMsgMsgIdEndpointGet(msgId, appId, iterator, limit, idempotencyKey, _options) {
        const result = this.api.listAttemptedDestinationsApiV1AppAppIdMsgMsgIdEndpointGet(msgId, appId, iterator, limit, idempotencyKey, _options);
        return result.toPromise();
    }
    listAttemptedMessagesApiV1AppAppIdEndpointEndpointIdMsgGet(endpointId, appId, iterator, limit, channel, status, before, after, idempotencyKey, _options) {
        const result = this.api.listAttemptedMessagesApiV1AppAppIdEndpointEndpointIdMsgGet(endpointId, appId, iterator, limit, channel, status, before, after, idempotencyKey, _options);
        return result.toPromise();
    }
    listAttemptsApiV1AppAppIdMsgMsgIdAttemptGet(appId, msgId, iterator, limit, endpointId, eventTypes, channel, status, before, after, idempotencyKey, _options) {
        const result = this.api.listAttemptsApiV1AppAppIdMsgMsgIdAttemptGet(appId, msgId, iterator, limit, endpointId, eventTypes, channel, status, before, after, idempotencyKey, _options);
        return result.toPromise();
    }
    listAttemptsByEndpointApiV1AppAppIdAttemptEndpointEndpointIdGet(appId, endpointId, iterator, limit, status, statusCodeClass, eventTypes, channel, before, after, idempotencyKey, _options) {
        const result = this.api.listAttemptsByEndpointApiV1AppAppIdAttemptEndpointEndpointIdGet(appId, endpointId, iterator, limit, status, statusCodeClass, eventTypes, channel, before, after, idempotencyKey, _options);
        return result.toPromise();
    }
    listAttemptsByMsgApiV1AppAppIdAttemptMsgMsgIdGet(appId, msgId, endpointId, iterator, limit, status, statusCodeClass, eventTypes, channel, before, after, idempotencyKey, _options) {
        const result = this.api.listAttemptsByMsgApiV1AppAppIdAttemptMsgMsgIdGet(appId, msgId, endpointId, iterator, limit, status, statusCodeClass, eventTypes, channel, before, after, idempotencyKey, _options);
        return result.toPromise();
    }
    listAttemptsForEndpointApiV1AppAppIdMsgMsgIdEndpointEndpointIdAttemptGet(msgId, appId, endpointId, iterator, limit, eventTypes, channel, status, before, after, idempotencyKey, _options) {
        const result = this.api.listAttemptsForEndpointApiV1AppAppIdMsgMsgIdEndpointEndpointIdAttemptGet(msgId, appId, endpointId, iterator, limit, eventTypes, channel, status, before, after, idempotencyKey, _options);
        return result.toPromise();
    }
    resendWebhookApiV1AppAppIdMsgMsgIdEndpointEndpointIdResendPost(endpointId, msgId, appId, idempotencyKey, _options) {
        const result = this.api.resendWebhookApiV1AppAppIdMsgMsgIdEndpointEndpointIdResendPost(endpointId, msgId, appId, idempotencyKey, _options);
        return result.toPromise();
    }
}
exports.PromiseMessageAttemptApi = PromiseMessageAttemptApi;
const ObservableAPI_11 = require("./ObservableAPI");
class PromiseStatisticsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_11.ObservableStatisticsApi(configuration, requestFactory, responseProcessor);
    }
    getAppAttemptStatsApiV1StatsAppAppIdAttemptGet(appId, startDate, endDate, idempotencyKey, _options) {
        const result = this.api.getAppAttemptStatsApiV1StatsAppAppIdAttemptGet(appId, startDate, endDate, idempotencyKey, _options);
        return result.toPromise();
    }
    getEpStatsApiV1StatsAppAppIdEpEndpointIdAttemptGet(endpointId, appId, startDate, endDate, idempotencyKey, _options) {
        const result = this.api.getEpStatsApiV1StatsAppAppIdEpEndpointIdAttemptGet(endpointId, appId, startDate, endDate, idempotencyKey, _options);
        return result.toPromise();
    }
}
exports.PromiseStatisticsApi = PromiseStatisticsApi;
