// @candypay/react-checkout-sdk/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  CandyPayContext: () => CandyPayContext,
  CandyPayProvider: () => CandyPayProvider,
  CheckoutButton: () => CheckoutButton,
  useCheckout: () => useCheckout
});
module.exports = __toCommonJS(src_exports);

// src/helpers/handleConversion.ts
var unit8ToString = (uint8) => {
  let binary = "";
  const bytes = [].slice.call(uint8);
  bytes.forEach((b) => binary += String.fromCharCode(b));
  return encodeURIComponent(window.btoa(binary));
};

// src/helpers/pako.ts
var import_pako = __toESM(require("pako"));
var compress = (data) => {
  return import_pako.default.deflate(JSON.stringify(data));
};

// src/lib/constants.ts
var REDIRECT_URL = "https://checkout.candypay.fun/";

// src/components/CheckoutBtn.tsx
var import_react = require("@chakra-ui/react");
var import_react2 = require("@stitches/react");
var import_react3 = require("react");

// src/components/SolanaLogo.tsx
var import_jsx_runtime = require("react/jsx-runtime");
var SolanaLogo = ({}) => {
  return /* @__PURE__ */ (0, import_jsx_runtime.jsxs)("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    width: "20",
    height: "18",
    viewBox: "0 0 20 18",
    children: [
      /* @__PURE__ */ (0, import_jsx_runtime.jsx)("defs", {
        children: /* @__PURE__ */ (0, import_jsx_runtime.jsxs)("linearGradient", {
          id: "linear0",
          x1: "8.526",
          x2: "88.993",
          y1: "90.097",
          y2: "-3.016",
          gradientTransform: "scale(.19802 .20455)",
          gradientUnits: "userSpaceOnUse",
          children: [
            /* @__PURE__ */ (0, import_jsx_runtime.jsx)("stop", {
              offset: "0.08",
              stopColor: "#9945FF"
            }),
            /* @__PURE__ */ (0, import_jsx_runtime.jsx)("stop", {
              offset: "0.3",
              stopColor: "#8752F3"
            }),
            /* @__PURE__ */ (0, import_jsx_runtime.jsx)("stop", {
              offset: "0.5",
              stopColor: "#5497D5"
            }),
            /* @__PURE__ */ (0, import_jsx_runtime.jsx)("stop", {
              offset: "0.6",
              stopColor: "#43B4CA"
            }),
            /* @__PURE__ */ (0, import_jsx_runtime.jsx)("stop", {
              offset: "0.72",
              stopColor: "#28E0B9"
            }),
            /* @__PURE__ */ (0, import_jsx_runtime.jsx)("stop", {
              offset: "0.97",
              stopColor: "#19FB9B"
            })
          ]
        })
      }),
      /* @__PURE__ */ (0, import_jsx_runtime.jsx)("path", {
        fill: "url(#linear0)",
        d: "M19.898 14.191l-3.304 3.567a.789.789 0 01-.254.18.833.833 0 01-.305.062H.383a.39.39 0 01-.352-.23.386.386 0 01-.023-.22.378.378 0 01.094-.198l3.304-3.563a.7.7 0 01.258-.18.675.675 0 01.3-.066h15.653c.074 0 .145.02.207.062a.402.402 0 01.145.168c.031.07.039.149.023.22a.365.365 0 01-.094.198zM16.594 7.02a.754.754 0 00-.254-.188.833.833 0 00-.305-.062H.383a.39.39 0 00-.207.062.438.438 0 00-.145.172.379.379 0 00-.023.219c.012.074.043.14.094.199l3.304 3.558c.07.079.16.141.258.184.094.043.2.063.3.066h15.653a.39.39 0 00.207-.062.438.438 0 00.145-.172c.031-.07.039-.14.023-.219a.393.393 0 00-.094-.199zM.383 4.457h15.652c.102 0 .207-.02.305-.066a.685.685 0 00.254-.18L19.898.648a.378.378 0 00.094-.199.355.355 0 00-.023-.219.39.39 0 00-.352-.23H3.965a.797.797 0 00-.3.063.872.872 0 00-.259.18L.102 3.808a.38.38 0 00-.094.199.386.386 0 00.023.219.37.37 0 00.145.164.336.336 0 00.207.066zm0 0"
      })
    ]
  });
};
var SolanaLogo_default = SolanaLogo;

// src/components/CheckoutBtn.tsx
var import_jsx_runtime2 = require("react/jsx-runtime");
var globalStyles = (0, import_react2.globalCss)({
  "@import": "url('https://fonts.googleapis.com/css2?family=Red+Hat+Display:wght@500&display=swap')",
  img: {
    height: "1.3rem"
  }
});
var CheckoutButton = ({
  handleSession,
  children,
  ...props
}) => {
  const [isLoading, setIsLoading] = (0, import_react3.useState)(false);
  const { publicApiKey } = (0, import_react3.useContext)(CandyPayContext);
  globalStyles();
  const handleClick = async () => {
    setIsLoading(true);
    try {
      const sessionId = await handleSession();
      const compressed = compress({
        sessionId,
        publicApiKey
      });
      window.open(
        `${REDIRECT_URL}pay?data=${unit8ToString(compressed)}`,
        "_self"
      );
      setIsLoading(false);
    } catch (error) {
      setIsLoading(false);
    }
  };
  return /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(import_react.Button, {
    color: "white",
    fontWeight: "medium",
    w: "36",
    onClick: handleClick,
    bgColor: "#010101",
    _hover: { bgColor: "#1a1a1a" },
    isDisabled: !publicApiKey,
    fontFamily: "Red Hat Display, sans-serif",
    isLoading,
    ...props,
    display: "flex",
    alignItems: "center",
    gap: "2",
    children: children || /* @__PURE__ */ (0, import_jsx_runtime2.jsxs)(import_jsx_runtime2.Fragment, {
      children: [
        "Pay with ",
        /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(SolanaLogo_default, {})
      ]
    })
  });
};

// src/hooks/useCheckout.ts
var import_react4 = require("react");
var useCheckout = (genSession) => {
  const [isLoading, setLoading] = (0, import_react4.useState)(false);
  const { publicApiKey } = (0, import_react4.useContext)(CandyPayContext);
  const mutate = async () => {
    setLoading(true);
    const sessionId = await genSession();
    const compressed = compress({
      sessionId,
      publicApiKey
    });
    setLoading(false);
    window.open(
      `${REDIRECT_URL}/pay?data=${unit8ToString(compressed)}`,
      "_self"
    );
  };
  return { mutate, isLoading };
};

// src/providers/CandyPayProvider.tsx
var import_react5 = require("@chakra-ui/react");
var import_react6 = require("react");
var import_jsx_runtime3 = require("react/jsx-runtime");
var CandyPayContext = (0, import_react6.createContext)({
  publicApiKey: ""
});
var CandyPayProvider = ({ children, publicApiKey }) => {
  return /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(CandyPayContext.Provider, {
    value: { publicApiKey },
    children: /* @__PURE__ */ (0, import_jsx_runtime3.jsx)(import_react5.ChakraProvider, {
      children
    })
  });
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CandyPayContext,
  CandyPayProvider,
  CheckoutButton,
  useCheckout
});
