// @metaplex-foundation/js/dist/cjs/errors/MetaplexError.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../_virtual/_rollupPluginBabelHelpers.cjs');

/** @group Errors */
class MetaplexError extends Error {
  constructor(input) {
    super(input.problem);

    _rollupPluginBabelHelpers.defineProperty(this, "name", 'MetaplexError');

    this.key = `metaplex.errors.${input.key}`;
    this.title = input.title;
    this.problem = overrideWithOptions(input.problem, input.options?.problem, input.options?.problemPrefix, input.options?.problemSuffix);
    this.solution = overrideWithOptions(input.solution, input.options?.solution, input.options?.solutionPrefix, input.options?.solutionSuffix);
    this.source = input.source;
    this.sourceDetails = input.sourceDetails;
    this.cause = input.options?.cause;
    this.logs = input.options?.logs;
    this.message = this.toString(false);
  }

  getCapitalizedSource() {
    if (this.source === 'sdk' || this.source === 'rpc') {
      return this.source.toUpperCase();
    }

    return this.source[0].toUpperCase() + this.source.slice(1);
  }

  getFullSource() {
    const capitalizedSource = this.getCapitalizedSource();
    const sourceDetails = this.sourceDetails ? ` > ${this.sourceDetails}` : '';
    return capitalizedSource + sourceDetails;
  }

  toString(withName = true) {
    const logs = this.logs != null ? `\n\n[ Logs: ${this.logs.join(' |$> ')} ]` : '';
    const causedBy = this.cause ? `\n\nCaused By: ${this.cause}` : '';
    return (withName ? `[${this.name}] ` : '') + `${this.title}` + `\n>> Source: ${this.getFullSource()}` + `\n>> Problem: ${this.problem}` + `\n>> Solution: ${this.solution}` + causedBy + logs + '\n';
  }

}

const overrideWithOptions = (defaultText, override, prefix, suffix) => {
  return [prefix, override ?? defaultText, suffix].filter(text => !!text).join(' ');
};

exports.MetaplexError = MetaplexError;
